lib/streamlit/config.py

```diff
@@ -514,6 +514,18 @@ def _logger_message_format() -> str:
     type_=bool,
 )
 
+_create_option(
+    "runner.enforceSerializableSessionState",
+    description="""
+        Warn about adding unserializable data to Session State. Some execution
+        environments may require serializing all data in Session State, so
+        it may be useful to detect incompatibility during development, or
+        when the execution environment will stop supporting it in the future.
+    """,
```

Comment:
This looks great - thank you! ✅

Reply:
Actually 😅 - it says "Warn", the actual behavior is to raise an exception (which will halt execution and make a red box) is that right?

If so, maybe it should be "Raise an exception when adding unserializable..."? or something like that?